### PR TITLE
feat: implement indexing

### PR DESCRIPTION
**Pull request by muralov**
**Description**

Changes proposed in this pull request:

- Implement loading markdown files
- Implement chunking markdown files
- Implement saving the embeddings for the chunks in Hana DB
- Implement unit and integration tests

**Related issue(s)**
#199

### CODE REVIEW

**Review by the1bit — APPROVED** [review]

File: doc_indexer/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+1. Install the dependencies:
+
+```bash
+poetry install
+```
+
+2. Run the project:
+
+Create .env file in the root of the project and add the following variables:
```
Comment (grego952) [inline doc_indexer/README.md:24]:
```suggestion
Create the `.env` file in the root of the project and add the following variables:
```

Comment (grego952) [inline doc_indexer/README.md:24]: I'd mark this line as step 2 and then the `Run the project` line as step 3.

File: doc_indexer/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+It is important to pay attention that `DOCS_PATH` is the path to the Kyma documentation markdown files.
+
+```bash
+poetry run python src/main.py
+```
+
+## Testing
+
+The .env can be used for testing as well. You can also create a separate .env.test with similar content to .env.
```
Comment (grego952) [inline doc_indexer/README.md:55]:
```suggestion
The `.env` file can also be used for testing. You can create a separate `.env.test` file with similar content to `.env`.
```

File: doc_indexer/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+poetry run poe test-unit
+```
+
+To run the integration tests, use the following command:
+
+```bash
+poetry run poe test-integration
+```
+
+## Static code analysis
```
Comment (grego952) [inline doc_indexer/README.md:75]:
```suggestion
## Static Code Analysis
```

**Review by grego952 — APPROVED** [review]

**Review by the1bit — APPROVED** [review]

**Pull request merged**